### PR TITLE
Align project structure with other server implementations

### LINKED ISSUE #29: Restructure Mesos projects
As a developer, I'd like to emulate the restructuring of `Local` Server and adapt the `Mesos` implementation, so they're all aligned with consistent project structure/assembly.

**Acceptance:**
Following three projects are available
- spring-cloud-dataflow-deployer-mesos
- spring-cloud-starter-dataflow-server-mesos
- spring-cloud-dataflow-server-mesos

_note: depending on the naming decision [i.e. `spring-cloud-dataflow-deployer-mesos` vs. `spring-cloud-mesos`], we may have to host the `mesos` deployer in its own repo_

### PR DESCRIPTION
**Pull request by trisberg**
Fixes #29

### CODE REVIEW

File: spring-cloud-dataflow-deployer-mesos/src/main/java/org/springframework/cloud/dataflow/server/spi/marathon/MarathonAutoConfiguration.java [hunk 1]
```diff
@@ -14,14 +14,14 @@
...
-package org.springframework.cloud.dataflow.admin.spi.marathon;
+package org.springframework.cloud.dataflow.server.spi.marathon;
 
 import org.springframework.boot.autoconfigure.AutoConfigureBefore;
 import org.springframework.boot.autoconfigure.data.redis.RedisAutoConfiguration;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.cloud.Cloud;
 import org.springframework.cloud.CloudFactory;
-import org.springframework.cloud.dataflow.admin.config.AdminProperties;
+import org.springframework.cloud.dataflow.server.config.DataFlowServerProperties;
```
Comment (markfisher) [inline spring-cloud-dataflow-deployer-mesos/src/main/java/org/springframework/cloud/dataflow/server/spi/marathon/MarathonAutoConfiguration.java]: these are not ordered alphabetically now

File: spring-cloud-dataflow-server-mesos/pom.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	<packaging>jar</packaging>
+	<description>Mesos Data Flow Server</description>
+	<parent>
+		<groupId>org.springframework.cloud</groupId>
+		<artifactId>spring-cloud-dataflow-server-mesos-parent</artifactId>
+		<version>1.0.0.BUILD-SNAPSHOT</version>
+	</parent>
+	<properties>
+		<start-class>org.springframework.cloud.dataflow.server.mesos.MesosDataFlowServer</start-class>
+		<!-- The following marathon connector version property is inserted in server.yml -->
```
Comment (markfisher) [inline spring-cloud-dataflow-server-mesos/pom.xml]: should be `dataflow-server.yml` in this comment

File: spring-cloud-dataflow-deployer-mesos/src/main/java/org/springframework/cloud/dataflow/server/spi/marathon/MarathonAutoConfiguration.java (hunk 1 above)
Comment (trisberg) [inline spring-cloud-dataflow-deployer-mesos/src/main/java/org/springframework/cloud/dataflow/server/spi/marathon/MarathonAutoConfiguration.java]: fixing

**Comment by markfisher** [comment]
squashed and merged as https://github.com/spring-cloud/spring-cloud-dataflow-admin-mesos/commit/2c6008552f9f109b90a32ab66b269d85e195a056

**Pull request closed without merging**